### PR TITLE
Added Chapter Nine

### PR DESCRIPTION
**Pull request by MarkWignall**

### CODE REVIEW

File: chapter09/mdw/Ex04.pl [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/perl
+
+use strict;
+use warnings;
+
+$^I= ".out";
+
+my $first = 1;
```
Comment (thatguy59) [inline chapter09/mdw/Ex04.pl:8]: I don't think this will work over multiple files, because this won't be reset when we open a new one.

Comment (MarkWignall) [inline chapter09/mdw/Ex04.pl:8]: Yeah, I should have added a line after the print to reset $first if /\Z/, or something to that effect.

File: chapter09/mdw/Ex05.pl [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+$^I= ".out";
+
+my $line = 1;
+
+while (<>) {
+	if ( $line == 2 ) {
+		if ( !(/\#\# Copyright \(C\) 2015 by mdw/) ){
+			s/(.*)/## Copyright (C) 2015 by mdw\n$1/;
+		};
```
Comment (pratyp) [inline chapter09/mdw/Ex05.pl:14]: So I'm assuming the extra <code>;</code> after <code>if</code> block is to say there is no <code>else</code> block?

Comment (MarkWignall) [inline chapter09/mdw/Ex05.pl:14]: Yeah, I don't know why I put the ";" there. I don't think it affects how code is run it just a style thing. I don't know if there is a recommendation on if it is helpful/confusing.

File: chapter09/mdw/Ex05.pl [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+use strict;
+use warnings;
+
+$^I= ".out";
+
+my $line = 1;
+
+while (<>) {
+	if ( $line == 2 ) {
+		if ( !(/\#\# Copyright \(C\) 2015 by mdw/) ){
```
Comment (qianliok) [inline chapter09/mdw/Ex05.pl:12]: could rewrite as `$line == 2 &&  !(/\#\# Copyright \(C\) 2015 by mdw/)`, but I'm still not sure about hardcode with line 2, what if the copyright is at line 3?

**Pull request merged**